pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -33,15 +41,15 @@ func (ca *cloudCIDRAllocator) PerformMultiNetworkCIDRAllocation(node *v1.Node, i
 			rangeNameAliasIPMap[ipRange.SubnetworkRangeName] = ipRange
 		}
 		for _, network := range networks {
-			klog.V(4).Infof("allotting pod cidrs for network %s", network.Name)
+			klog.V(4).InfoS("allotting pod CIDRs", "nodeName", node.Name, "networkName", network.Name)
 			gnp, err := ca.gnpLister.Get(network.Spec.ParametersRef.Name)
 			if err != nil {
-				return nil, nil, nil, err
+				return nil, err
```

Comment:
if the gnpObject doesn't exist do you want to return or to continue?
if err != nil {
   if apierrors.IsNotFound(err) {
        klog(not found %s=
        continue
   } else {
       return err, nil
}
```

Reply:
good catch, this should continue, but I am aligning next PR after this, which filters Networks based on Ready Condition. If a Network is Ready and GNP cannot be found that should error. so will keep this as is.